### PR TITLE
Add selectionStart and selectionEnd variables to markdown input ref 

### PR DESCRIPTION
**Pull request by Skalakid**
### Details

This PR fixes how our markdown text input tracks text selection and adds `selectionStart` and `selectionEnd` variables to input ref.

### Related Issues

https://github.com/Expensify/react-native-live-markdown/issues/173

### Manual Tests

Using reproduction code from the linked issue start changing the text selection. Verify in console if it returns correct values. Next, position your cursor at the beginning of the markdown text input and after clicking `arrow up`, verify if focus is changed into RN text input that is above

### Linked PRs

https://github.com/Expensify/App/issues/36173

### CODE REVIEW

**Review by robertKozik — APPROVED** [review]

File: src/MarkdownTextInput.web.tsx [hunk 1]
```diff
@@ -167,6 +167,17 @@ const MarkdownTextInput = React.forwardRef<TextInput, MarkdownTextInputProps>(
...
     const heightSafePlaceholder = useMemo(() => getPlaceholderValue(placeholder), [placeholder]);
 
+    const updateSelection = useCallback(() => {
+      if (!divRef.current) {
+        return;
+      }
+      const selection = CursorUtils.getCurrentCursorPosition(divRef.current);
+      contentSelection.current = selection;
+      (divRef.current as HTMLInputElement).selectionStart = selection.start;
+      (divRef.current as HTMLInputElement).selectionEnd = selection.end;
```
Comment (tomekzaw) [inline src/MarkdownTextInput.web.tsx:177]: Let's assign `divRef.current` to a new variable

File: src/MarkdownTextInput.web.tsx [hunk 2]
```diff
@@ -167,6 +167,18 @@ const MarkdownTextInput = React.forwardRef<TextInput, MarkdownTextInputProps>(
...
+    const updateSelection = useCallback(() => {
+      if (!divRef.current) {
+        return;
+      }
+      const selection = CursorUtils.getCurrentCursorPosition(divRef.current);
+      const markdownHTMLInput = divRef.current as HTMLInputElement;
+      markdownHTMLInput.selectionStart = selection.start;
+      markdownHTMLInput.selectionEnd = selection.end;
+      contentSelection.current = selection;
+      return selection;
```
Comment (tomekzaw) [inline src/MarkdownTextInput.web.tsx:179]:
```suggestion
```

File: src/MarkdownTextInput.web.tsx [hunk 3]
```diff
@@ -350,13 +357,16 @@ const MarkdownTextInput = React.forwardRef<TextInput, MarkdownTextInputProps>(
       [onKeyPress],
     );
 
+    const handleKeyUp = useCallback(() => {
+      updateSelection();
+    }, []);
+
```
Comment (tomekzaw) [inline src/MarkdownTextInput.web.tsx:363]:
```suggestion
```

File: src/MarkdownTextInput.web.tsx [hunk 4]
```diff
@@ -350,13 +357,16 @@ const MarkdownTextInput = React.forwardRef<TextInput, MarkdownTextInputProps>(
...
     const handleSelectionChange: ReactEventHandler<HTMLDivElement> = useCallback(
       (event) => {
         const e = event as unknown as NativeSyntheticEvent<TextInputSelectionChangeEventData>;
         setEventProps(e);
-        const selection = CursorUtils.getCurrentCursorPosition(e.target as unknown as HTMLElement);
-        contentSelection.current = selection;
-        if (onSelectionChange) {
+        const selection = updateSelection();
+        if (onSelectionChange && selection) {
```
Comment (tomekzaw) [inline src/MarkdownTextInput.web.tsx:369]:
```suggestion
        updateSelection();
        if (onSelectionChange && contentSelection.current) {
```

File: src/MarkdownTextInput.web.tsx [hunk 5]
```diff
@@ -488,6 +499,7 @@ const MarkdownTextInput = React.forwardRef<TextInput, MarkdownTextInputProps>(
         autoCapitalize={autoCapitalize}
         className={className}
         onKeyDown={handleKeyPress}
+        onKeyUp={handleKeyUp}
```
Comment (tomekzaw) [inline src/MarkdownTextInput.web.tsx:502]:
```suggestion
        onKeyUp={updateSelection}
```

**Review by tomekzaw — CHANGES_REQUESTED** [review]

**Review by tomekzaw — APPROVED** [review]
Thanks for fixing this!

**Pull request merged**